packages/client-web/src/connection/web_connection.ts

```diff
@@ -69,15 +71,16 @@ export class WebConnection implements Connection<ReadableStream> {
     return {
       query_id: result.query_id,
       stream: result.stream || new ReadableStream<Uint8Array>(),
+      response_headers: result.response_headers,
```

Comment:
It's the first HTTP abstraction that leaks in the result set. I'd rather add some generic method to avoid a breaking change once TCP protocol is supported, but your call

Reply:
I think it is already leaking a bit, as we allow `text/`json` calls, which are not very compatible with Native and even RowBinary format; in time, we will likely have to implement the native version with a new result set flavor (similarly to RowBinary).